index.bs

```diff
@@ -1715,3 +1705,220 @@ When [=a browsing context is discarded=], run the following steps:
+```
+Realm = text;
+
+RealmOrBrowsingContext = (
+  {realm: Realm} //
+  {context: BrowsingContext}
+)
+
+RealmInfo = {
+  realm: Realm,
+  type: RealmType,
+  origin: text,
+}
+
+RealmType = (
+  "window" / "worker" / "sharedworker" / "serviceworker" / "worklet" / text
+)
+```
+
+Each [=realm=] has an associated <dfn export>realm id</dfn>, which is a string
+uniquely identifying that realm. This is implicitly set when the realm is
+created.
+
+<div algorithm>
+To <dfn>get a realm</dfn> given |realm id|:
+
+ 1. If |realm id| is null, return [=success=] with data null.
+
+ 1. If there is no realm with [=realm id=] |realm id| return
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |realm| be the browsing context with id |realm id|.
+
+ 1. Return [=success=] with data |realm|
+
+Issue: This has the wrong error code
+
+</div>
+
+<div algorithm>
+To <dfn>get the realm info</dfn> given |environment settings|:
+
+ 1. Let |realm| be |environment settings|' [=realm execution context=]'s Realm component.
+
+ 1. Let |realm id| be the [=realm id=] for |realm|.
+
+ 1. Run the steps under the first matching condition:
+
+   <dl>
+     <dt>The [=Realm/global object=] specified by |environment settings| is a [=Window=] object
+     <dd>
+      1. If [=Realm/global object=] specified by |environment settings| has a [=Document=]
+         that is not yet fully active, return null.
+
+      1. Let |type| be "<code>window</code>".
+
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{DedicatedWorkerGlobalScope}} object
+     <dd>
+       1. Let |type| be "<code>worker</code>".
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{SharedWorkerGlobalScope}} object
+
+    <dd>
+       1. Let |type| be "<code>sharedworker</code>".
+     <dt>The [=Realm/global object=] specified by |environment settings| is a {{ServiceWorkerGlobalScope}} object
+
+    <dd>
+       1. Let |type| be "<code>serviceworker</code>".
+
+    <dt>The [=Realm/global object=] specified by |environment settings| is a {{WorkletGlobalScope}} object
+    <dd>
+       1. Let |type| be "<code>worklet</code>".
+
+    <dt>Otherwise:
+    <dd>
+      1. Let |type| be an implementation-defined string, consisting of a prefix,
+         followed by a colon ("<code>:</code>") followed by a type name.
+   </dl>
+
+ 1. Let |origin| be the [=serialization of an origin=] given |environment settings|'s |origin|.
+
+ 1. Let |realm info| be a map matching the <code>RealmInfo</code> production,
+    with the <code>realm</code> field set to |realm id|, the <code>type</code>
+    field set to |type| and the <code>origin</code> field set to |origin|.
+
+ 1. Return |realm info|
+
+</div>
+
+### Commands ### {#module-script-commands}
+
+#### getRealms ####  {#command-script-getrealms}
+
+The <dfn export for=commands>getRealms</dfn> command returns a list of all
+realms, optionally filtered to realms of a specific type, or to the realm
```

Comment:
```suggestion
realms, optionally filtered to [=realms=] of a specific type, or to the realm
```

Perhaps we can also define type separately, so that it's linkable here?